subprojects/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -221,6 +233,42 @@ private static void validatePackageName(String packageName) {
         }
     }
 
+    static class JavaVersionSelection {
+        private final JavaLanguageVersion languageVersion;
+
+        JavaVersionSelection(JavaLanguageVersion ver) {
+            languageVersion = ver;
+        }
+
+        @Override
+        public String toString() {
+            return "Java " + languageVersion;
+        }
+
+        public JavaLanguageVersion getLanguageVersion() {
+            return languageVersion;
+        }
+    }
+
+    java.util.Optional<JavaLanguageVersion> getJavaLanguageVersion(boolean useIncubatingAPIs, UserInputHandler inputHandler, Language language) {
+        if (isJvmLanguage(language) && useIncubatingAPIs) {
```

Comment:
I think the way we've done the other questions is to ask the init descriptor what it supports. I think we should do something similar here.